inst/pages/multimedia.qmd

```diff
@@ -0,0 +1,589 @@
+  M3 -> Outcome
+  M4 -> Outcome
+  Exposure -> Outcome [style=dashed]
+}
+")
+```
+
+In this chapter, we demonstrate multimodal mediation analysis using microbiome 
+data from the iHMP IBDMDB study from the R/Bioconductor package 
+curatedMetagenomicData. [@Lloyd-Price2019]. Unlike conventional mediation 
+analysis, which typically focuses on a single mediator or a small set of 
+mediators, we performed multimodal mediation analysis with R/Bioconductor 
+package \[multimedia\] [@Jiang2025] here, which can handle many potential 
+mediators across multiple data modalities. For example, species-level taxonomic 
+abundances, pathways abundances, and metabolomic profiles can all serve as 
+potential mediators. This method allows us to estimate and rank each mediator’s 
+contribution while accounting for correlations between mediators in a high- 
+dimensional setting.
+
+In this example, time point (baseline vs. post-treatment) is used as the 
+treatment, and microbiome-based dysbiosis scores are used as the outcome. We 
+will first focus on the relative abundances of gut microbial species as 
+mediators, followed by pathways abundances.
+
+Generally, we will proceed through the following key steps:
+
+1.  Prepare the microbiome taxonomic data from the iHMP IBDMDB project.
+
+2.  Calculate dysbiosis scores as the outcome.
+
+3.  Define the mediation analysis data structure.
+
+4.  Fit the multimodal mediation model using the R package multimedia.
+
+5.  Interpret both the overall indirect effects and the mediator-specific 
+indirect effects.
+
+6.  Visualize the results using forest plots, histograms, and rankings to 
+prioritize findings.
+
+7.  Repeat the process for the iHMP microbial pathways.
+
+### Performing multimodal mediation analysis for iHMP species relative abundance {#sec-relative-abundance}
+
+We begin by loading the relative abundance data and calculating dysbiosis scores 
+for each sample based on Bray–Curtis dissimilarity from a healthy reference set. 
+This represents deviation from a healthy microbiome, where higher dysbiosis 
+scores indicate greater microbial imbalance. We then subset to IBD subjects with 
+samples collected at both baseline (visit 1) and post-treatment (visit 21).
+
+```{r}
+#| label: iHMP_data_preprocessing
+#| message: false
+
+##################
+# Load libraries #
+##################
+
+library(curatedMetagenomicData)
+library(SummarizedExperiment)
+library(dplyr)
+library(vegan)
+library(tidyverse)
+library(multimedia)
+library(stringr)
+library(ggplot2)
+library(mia)
+
+##################
+# Load iHMP data #
+##################
+
+# Load data
+tse <- curatedMetagenomicData(
+  "HMP_2019_ibdmdb.relative_abundance", 
+  rownames = "short",
+  dryrun = FALSE
+  )[[1]]
+
+# Assign SampleID for matching
+tse[["SampleID"]] <- colnames(tse)
+
+# Convert relative_abundance assay to relabundance (which is in [0,1] interval)
+tse <- transformAssay(tse, assay.type="relative_abundance", method="relabundance")
```

Comment:
Would it be useful to include an explanation what is the actual difference between relative_abundance and relabundance? (not obvious to me at least by just reading the code)